Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -4849,11 +4849,11 @@ AccessGenerationResult InlineCacheCompiler::compile(const GCSafeConcurrentJSLock
 
     if (ASSERT_ENABLED) {
         if (m_stubInfo.useDataIC) {
-            jit.loadPtr(CCallHelpers::Address(GPRInfo::jitDataRegister, BaselineJITData::offsetOfStackOffset()), jit.scratchRegister());
-            jit.addPtr(jit.scratchRegister(), GPRInfo::callFrameRegister, jit.scratchRegister());
+            jit.loadPtr(CCallHelpers::Address(GPRInfo::jitDataRegister, BaselineJITData::offsetOfStackOffset()), m_scratchGPR);
+            jit.addPtr(jit.scratchRegister(), GPRInfo::callFrameRegister, m_scratchGPR);
```

Comment:
Looks incorrect as it is using unset scratchRegister

Reply:
It gets set above, right?
```
m_scratchGPR = allocator.allocateScratchGPR();
```

Reply:
Fixing this fixes a bunch of the failing tests

Reply:
Hmm I thought is was set here: https://github.com/WebKit/WebKit/pull/40966/files/53b5ec76c64b7979c2bfac0d7e5049d594ea7909#diff-28a2431a26da60c36cfb143a33266acc988650dbf6e5a0e6a8dabc50afca3fa2R4806

Reply:
```
 jit.addPtr(jit.scratchRegister(), 
 ```

Reply:
D'oh now I see it....